common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -750,6 +753,81 @@ public AccountDeletionRecord removeAccount(@Nullable final String environment,
         );
     }
 
+    /**
+     * Tests if a clientId is 'known' to the cache. A clientId is known if a token has been
+     * previously saved to the cache with it.
+     *
+     * @param clientId The clientId whose known status should be queried.
+     * @return True if this clientId is known. False otherwise.
+     */
+    public boolean isClientIdKnownToCache(@NonNull final String clientId) {
+        return getAllClientIds().contains(clientId);
+    }
+
+    /**
+     * Returns the List of FoCI users in the cache. This API is provided so that the broker may
+     * **internally** query the cache for known users, such that the broker may verify an
+     * unknown clientId is a part of the FoCI family.
+     * <p>
+     * Please note, the ICacheRecords returned by this query are NOT fully populated. Only the
+     * {@link GenericAccount} and {@link GenericRefreshToken} will be returned.
+     * will be resutned.
+     *
+     * @param environment The environment whose accounts should be queried.
+     * @return A List of ICacheRecords for the FoCI accounts.
+     */
+    @SuppressWarnings("unchecked")
+    public List<ICacheRecord> getFociCacheRecords(@NonNull final String environment) {
```

Comment:
Do we need to set the `environment` `nonnull` here? In `BrokerOAuth2TokenCache#getAccounts(java.lang.String, java.lang.String)`, `environment` is nullable.